tests/components/zha/test_config_flow.py

```diff
@@ -1,125 +1,237 @@
-    flow.hass = hass
+def com_port():
+    """Mock of a serial port."""
+    port = serial.tools.list_ports_common.ListPortInfo()
+    port.serial_number = "1234"
+    port.manufacturer = "Virtual serial port"
+    port.device = "/dev/ttyUSB1234"
+    port.description = "Some serial port"
 
-    with tests.async_mock.patch(
-        "homeassistant.components.zha.config_flow.check_zigpy_connection",
-        return_value=False,
-    ):
-        result = await flow.async_step_user(
-            user_input={"usb_path": "/dev/ttyUSB1", "radio_type": "ezsp"}
-        )
+    return port
 
-    assert result["errors"] == {"base": "cannot_connect"}
 
-    with tests.async_mock.patch(
-        "homeassistant.components.zha.config_flow.check_zigpy_connection",
-        return_value=True,
-    ):
-        result = await flow.async_step_user(
-            user_input={"usb_path": "/dev/ttyUSB1", "radio_type": "ezsp"}
-        )
+@patch("serial.tools.list_ports.comports", MagicMock(return_value=[com_port()]))
+@patch(
+    "homeassistant.components.zha.config_flow.detect_radios",
+    return_value={CONF_RADIO_TYPE: "test_radio"},
+)
+async def test_user_flow(detect_mock, hass):
+    """Test user flow -- radio detected."""
 
-    assert result["type"] == "create_entry"
-    assert result["title"] == "/dev/ttyUSB1"
-    assert result["data"] == {"usb_path": "/dev/ttyUSB1", "radio_type": "ezsp"}
+    port = com_port()
+    port_select = f"{port}, s/n: {port.serial_number} - {port.manufacturer}"
 
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={CONF_SOURCE: SOURCE_USER},
+        data={zigpy.config.CONF_DEVICE_PATH: port_select},
+    )
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"].startswith(port.description)
+    assert result["data"] == {CONF_RADIO_TYPE: "test_radio"}
+    assert detect_mock.await_count == 1
+    assert detect_mock.await_args[0][0] == port.device
+
+
+@patch("serial.tools.list_ports.comports", MagicMock(return_value=[com_port()]))
+@patch(
+    "homeassistant.components.zha.config_flow.detect_radios", return_value=None,
+)
+async def test_user_flow_not_detected(detect_mock, hass):
+    """Test user flow, radio not detected."""
+
+    port = com_port()
+    port_select = f"{port}, s/n: {port.serial_number} - {port.manufacturer}"
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={CONF_SOURCE: SOURCE_USER},
+        data={zigpy.config.CONF_DEVICE_PATH: port_select},
+    )
 
-async def test_user_flow_existing_config_entry(hass):
-    """Test if config entry already exists."""
-    MockConfigEntry(domain=DOMAIN, data={"usb_path": "/dev/ttyUSB1"}).add_to_hass(hass)
-    flow = config_flow.ZhaFlowHandler()
-    flow.hass = hass
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "pick_radio"
+    assert detect_mock.await_count == 1
+    assert detect_mock.await_args[0][0] == port.device
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)